### ISSUE TITLE
npc variable are shared by all his duplicated npc

### ISSUE THREAD

**Issue by vykimo** [HerculesWS/Hercules#1168]
```
> 2@bif,22,28,0	script	system	1_ETC_01,{
> mes ".variable is set to "+.variable;
> close2;
> .variable = 1;
> end;
> }
> 2@bif,20,28,0	duplicate(system)	system2	1_ETC_01
```
Try this npc, if you speak to the first one. he'll say ".variable is set to 0"
And if you speak now to the second one, he'll say ".variable is set to 1"
It's absolutely not normal, it is said in the doc that each npc has its own npc variable '.'

And for example, if I do this code, I don't have any problem...

```
> 2@bif,22,28,0	script	system	1_ETC_01,{
> mes ".variable is set to "+.variable;
> close2;
> .variable = 1;
> end;
> }
> 2@bif,20,28,0	script	system2	1_ETC_01,{
> mes ".variable is set to "+.variable;
> close2;
> .variable = 1;
> end;
> }
```

**Comment by dastgir** [comment]
Ok, thats not documented, but it's been like this since ages, and is the intended behavior.
Duplicate npc shared npc variable...

**Comment by boboka** [comment]
malamang duplicate bobo ka naman tanga

**Comment by MishimaHaruna** [comment]
Yes, it's by design. A very terrible design in my opinion (encourages the use of very ugly hacks in order to separate the variables), but still by design.

I intend on changing this with the new scripting engine in the future, but I don't have an estimate of when it'll be available.

**Comment by vykimo** [comment]
Thanks for informations. I agree with MishimaHaruna, totally useless thing to share npc variable but if some npc already use it.... I'm curious to know which one?
I must be prudent when using duplicata then... Maybe it could be great to specify this big information in script doc...?

**Comment by Ragno1** [comment]
Woe scripts make shared npc variables using a combination of setd + strnpcinfo to make an exclusive variable for every duplicated npc.

**Comment by Aeromesi** [comment]
Yes I came across this when I made one of my instances. What if there could be two commands like

"sharedduplicatenpc"

"noshareduplicatenpc"

That way we could choose which duplication process for whether or not we want variables sent to the other duplicate npc or not.
